conda/testing/helpers.py

```diff
@@ -296,7 +297,7 @@ def _get_index_r_base(
         ):
             sd._process_raw_repodata_str(json.dumps(repodata))
         sd._loaded = True
-        SubdirData._cache_[channel.url(with_credentials=True)] = sd
+        SubdirData._cache_[(channel.url(with_credentials=True), REPODATA_FN)] = sd
```

Comment:
Seems like a good addition to the cache

Reply:
The cache key changed a while ago but this was missed, I'm just updating it so we actually override the cache as intended.